4.5.1: build fails with gcc 11.0.0

gcc 11.0.0 from Fedora rawhide
```
make  -f modules/gapi/CMakeFiles/opencv_test_gapi.dir/build.make modules/gapi/CMakeFiles/opencv_test_gapi.dir/build
make[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu'
[ 69%] Building CXX object modules/gapi/CMakeFiles/opencv_test_gapi.dir/test/gapi_async_test.cpp.o
cd /home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu/modules/gapi && /usr/bin/g++ -DVK_NO_PROTOTYPES -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__OPENCV_TESTS=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu -I/usr/include/vulkan -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/ts/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/gapi/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/imgcodecs/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/videoio/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/core/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/flann/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/imgproc/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/dnn/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/features2d/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/calib3d/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/video/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/highgui/include -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/gapi/src -I/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu/3rdparty/ade/ade-0.1.1f/sources/ade/include -isystem /usr/include/gdal -isystem /usr/include/coin -isystem /usr/include/openblas -isystem /usr/include/eigen3 -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -flto=auto -flto-partition=none -fdata-sections -ffunction-sections   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -fopenmp -O2 -g -DNDEBUG -fPIE -std=c++11 -o CMakeFiles/opencv_test_gapi.dir/test/gapi_async_test.cpp.o -c /home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/gapi/test/gapi_async_test.cpp
/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/gapi/test/gapi_async_test.cpp: In member function ‘virtual void opencv_test::gtest_case_output_args_lifetime_::callback<gtest_TypeParam_>::TestBody()’:
/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/modules/gapi/test/gapi_async_test.cpp:448:26: error: ‘sleep_for’ is not a member of ‘std::this_thread’
  448 |        std::this_thread::sleep_for(std::chrono::milliseconds{2});
      |                          ^~~~~~~~~
make[2]: *** [modules/gapi/CMakeFiles/opencv_test_gapi.dir/build.make:293: modules/gapi/CMakeFiles/opencv_test_gapi.dir/test/gapi_async_test.cpp.o] Error 1
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu'
make[1]: *** [CMakeFiles/Makefile2:15960: modules/gapi/CMakeFiles/opencv_test_gapi.dir/all] Error 2
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu'
make: *** [Makefile:185: all] Error 2
```
```
$ cmake -L
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


CMake Error: The source directory "/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
-- Cache values
ANT_EXECUTABLE:FILEPATH=/bin/ant
BUILD_CUDA_STUBS:BOOL=OFF
BUILD_DOCS:BOOL=ON
BUILD_EXAMPLES:BOOL=ON
BUILD_IPP_IW:BOOL=ON
BUILD_ITT:BOOL=ON
BUILD_JASPER:BOOL=OFF
BUILD_JAVA:BOOL=ON
BUILD_JPEG:BOOL=OFF
BUILD_LIST:STRING=
BUILD_OPENEXR:BOOL=OFF
BUILD_OPENJPEG:BOOL=OFF
BUILD_PACKAGE:BOOL=ON
BUILD_PERF_TESTS:BOOL=ON
BUILD_PNG:BOOL=OFF
BUILD_PROTOBUF:BOOL=OFF
BUILD_SHARED_LIBS:BOOL=ON
BUILD_TBB:BOOL=OFF
BUILD_TESTS:BOOL=ON
BUILD_TIFF:BOOL=OFF
BUILD_USE_SYMLINKS:BOOL=OFF
BUILD_WEBP:BOOL=OFF
BUILD_WITH_DEBUG_INFO:BOOL=OFF
BUILD_WITH_DYNAMIC_IPP:BOOL=OFF
BUILD_ZLIB:BOOL=OFF
BUILD_opencv_alphamat:BOOL=ON
BUILD_opencv_apps:BOOL=ON
BUILD_opencv_aruco:BOOL=ON
BUILD_opencv_bgsegm:BOOL=ON
BUILD_opencv_bioinspired:BOOL=ON
BUILD_opencv_calib3d:BOOL=ON
BUILD_opencv_ccalib:BOOL=ON
BUILD_opencv_core:BOOL=ON
BUILD_opencv_datasets:BOOL=ON
BUILD_opencv_dnn:BOOL=ON
BUILD_opencv_dnn_objdetect:BOOL=ON
BUILD_opencv_dnn_superres:BOOL=ON
BUILD_opencv_dpm:BOOL=ON
BUILD_opencv_face:BOOL=ON
BUILD_opencv_features2d:BOOL=ON
BUILD_opencv_flann:BOOL=ON
BUILD_opencv_freetype:BOOL=ON
BUILD_opencv_fuzzy:BOOL=ON
BUILD_opencv_gapi:BOOL=ON
BUILD_opencv_hdf:BOOL=ON
BUILD_opencv_hfs:BOOL=ON
BUILD_opencv_highgui:BOOL=ON
BUILD_opencv_img_hash:BOOL=ON
BUILD_opencv_imgcodecs:BOOL=ON
BUILD_opencv_imgproc:BOOL=ON
BUILD_opencv_intensity_transform:BOOL=ON
BUILD_opencv_java:BOOL=ON
BUILD_opencv_java_bindings_generator:BOOL=ON
BUILD_opencv_js:BOOL=OFF
BUILD_opencv_js_bindings_generator:BOOL=ON
BUILD_opencv_line_descriptor:BOOL=ON
BUILD_opencv_mcc:BOOL=ON
BUILD_opencv_ml:BOOL=ON
BUILD_opencv_objc_bindings_generator:BOOL=ON
BUILD_opencv_objdetect:BOOL=ON
BUILD_opencv_optflow:BOOL=ON
BUILD_opencv_phase_unwrapping:BOOL=ON
BUILD_opencv_photo:BOOL=ON
BUILD_opencv_plot:BOOL=ON
BUILD_opencv_python3:BOOL=ON
BUILD_opencv_python_bindings_generator:BOOL=ON
BUILD_opencv_python_tests:BOOL=ON
BUILD_opencv_quality:BOOL=ON
BUILD_opencv_rapid:BOOL=ON
BUILD_opencv_reg:BOOL=ON
BUILD_opencv_rgbd:BOOL=ON
BUILD_opencv_saliency:BOOL=ON
BUILD_opencv_sfm:BOOL=ON
BUILD_opencv_shape:BOOL=ON
BUILD_opencv_stereo:BOOL=ON
BUILD_opencv_stitching:BOOL=ON
BUILD_opencv_structured_light:BOOL=ON
BUILD_opencv_superres:BOOL=ON
BUILD_opencv_surface_matching:BOOL=ON
BUILD_opencv_text:BOOL=ON
BUILD_opencv_tracking:BOOL=ON
BUILD_opencv_ts:BOOL=ON
BUILD_opencv_video:BOOL=ON
BUILD_opencv_videoio:BOOL=ON
BUILD_opencv_videostab:BOOL=ON
BUILD_opencv_world:BOOL=OFF
BUILD_opencv_xfeatures2d:BOOL=ON
BUILD_opencv_ximgproc:BOOL=ON
BUILD_opencv_xobjdetect:BOOL=ON
BUILD_opencv_xphoto:BOOL=ON
CCACHE_PROGRAM:FILEPATH=CCACHE_PROGRAM-NOTFOUND
CMAKE_BUILD_TYPE:STRING=RelWithDebInfo
CMAKE_CONFIGURATION_TYPES:STRING=Debug;Release
CMAKE_INSTALL_PREFIX:PATH=/usr
CPU_BASELINE:STRING=SSE3
CPU_DISPATCH:STRING=SSE4_1;SSE4_2;AVX;FP16;AVX2;AVX512_SKX
CV_DISABLE_OPTIMIZATION:BOOL=OFF
CV_ENABLE_INTRINSICS:BOOL=ON
CV_TRACE:BOOL=ON
Caffe_INCLUDE_DIR:PATH=Caffe_INCLUDE_DIR-NOTFOUND
Caffe_LIBS:FILEPATH=Caffe_LIBS-NOTFOUND
Ceres_DIR:PATH=/usr/lib64/cmake/Ceres
ComputeCpp_DIR:PATH=ComputeCpp_DIR-NOTFOUND
ENABLE_BUILD_HARDENING:BOOL=OFF
ENABLE_CCACHE:BOOL=ON
ENABLE_CONFIG_VERIFICATION:BOOL=OFF
ENABLE_COVERAGE:BOOL=OFF
ENABLE_FAST_MATH:BOOL=OFF
ENABLE_FLAKE8:BOOL=ON
ENABLE_GNU_STL_DEBUG:BOOL=OFF
ENABLE_IMPL_COLLECTION:BOOL=OFF
ENABLE_INSTRUMENTATION:BOOL=OFF
ENABLE_LTO:BOOL=OFF
ENABLE_NOISY_WARNINGS:BOOL=OFF
ENABLE_OMIT_FRAME_POINTER:BOOL=ON
ENABLE_PIC:BOOL=ON
ENABLE_PRECOMPILED_HEADERS:BOOL=OFF
ENABLE_PROFILING:BOOL=OFF
ENABLE_PYLINT:BOOL=ON
ENABLE_SOLUTION_FOLDERS:BOOL=OFF
EXECUTABLE_OUTPUT_PATH:PATH=/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu/bin
Flake8_DIR:PATH=Flake8_DIR-NOTFOUND
GDAL_CONFIG:FILEPATH=/usr/bin/gdal-config
GDAL_INCLUDE_DIR:PATH=/usr/include/gdal
GDAL_LIBRARY:FILEPATH=/lib64/libgdal.so
GDCM_DIR:PATH=/usr/lib64/cmake/gdcm
GENERATE_ABI_DESCRIPTOR:BOOL=OFF
Gflags_DIR:PATH=/usr/lib64/cmake/gflags
Glog_LIBS:FILEPATH=/lib64/libglog.so
HDF5_C_LIBRARY_hdf5:FILEPATH=/lib64/libhdf5.so
INSTALL_BIN_EXAMPLES:BOOL=OFF
INSTALL_CREATE_DISTRIB:BOOL=OFF
INSTALL_C_EXAMPLES:BOOL=ON
INSTALL_PYTHON_EXAMPLES:BOOL=ON
INSTALL_TESTS:BOOL=OFF
INSTALL_TO_MANGLED_PATHS:BOOL=OFF
LAPACK_CBLAS_H:STRING=cblas.h
LAPACK_IMPL:STRING=OpenBLAS
LAPACK_INCLUDE_DIR:PATH=/usr/include/openblas
LAPACK_LAPACKE_H:STRING=lapacke.h
LAPACK_LIBRARIES:STRING=/lib64/libopenblas.so
MFX_INCLUDE:PATH=MFX_INCLUDE-NOTFOUND
MFX_LIBRARY:FILEPATH=MFX_LIBRARY-NOTFOUND
MKL_ROOT_DIR:PATH=MKL_ROOT_DIR-NOTFOUND
MKL_USE_SINGLE_DYNAMIC_LIBRARY:BOOL=OFF
MKL_WITH_OPENMP:BOOL=OFF
MKL_WITH_TBB:BOOL=OFF
OGRE_DIR:PATH=OGRE_DIR-NOTFOUND
OPENCV_CONFIG_FILE_INCLUDE_DIR:PATH=/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu
OPENCV_DNN_CUDA:BOOL=OFF
OPENCV_DNN_OPENCL:BOOL=OFF
OPENCV_DOCS_DOT_IMAGE_FORMAT:STRING=svg
OPENCV_DOCS_DOT_PATH:PATH=
OPENCV_DOCS_DOXYFILE_IN:PATH=Doxyfile.in
OPENCV_DOCS_DOXYGEN_LAYOUT:PATH=DoxygenLayout.xml
OPENCV_DOCS_HAVE_DOT:BOOL=NO
OPENCV_DOCS_INTERACTIVE_SVG:BOOL=NO
OPENCV_DOWNLOAD_PATH:PATH=/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/.cache
OPENCV_DUMP_HOOKS_FLOW:BOOL=OFF
OPENCV_ENABLE_ALLOCATOR_STATS:BOOL=ON
OPENCV_ENABLE_ATOMIC_LONG_LONG:BOOL=ON
OPENCV_ENABLE_MEMALIGN:BOOL=ON
OPENCV_ENABLE_MEMORY_SANITIZER:BOOL=OFF
OPENCV_ENABLE_NONFREE:BOOL=OFF
OPENCV_EXTRA_MODULES_PATH:PATH=/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/opencv_contrib-4.5.1/modules
OPENCV_FORCE_3RDPARTY_BUILD:BOOL=OFF
OPENCV_FORCE_PYTHON_LIBS:BOOL=OFF
OPENCV_GENERATE_PKGCONFIG:BOOL=ON
OPENCV_GENERATE_SETUPVARS:BOOL=ON
OPENCV_IPP_GAUSSIAN_BLUR:BOOL=OFF
OPENCV_JAVADOC_DESTINATION:STRING=/home/tkloczko/rpmbuild/BUILD/opencv-4.5.1/x86_64-redhat-linux-gnu/doc/doxygen/html/javadoc
OPENCV_JAVADOC_LINK_URL:STRING=
OPENCV_JAVA_SOURCE_VERSION:STRING=
OPENCV_JAVA_TARGET_VERSION:STRING=
OPENCV_MATHJAX_RELPATH:STRING=https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0
OPENCV_PYTHON3_VERSION:BOOL=OFF
OPENCV_TIMESTAMP:STRING=2021-01-02T02:43:22Z
OPENCV_WARNINGS_ARE_ERRORS:BOOL=OFF
OPENEXR_INCLUDE_PATH:PATH=/usr/include/OpenEXR
OpenCV_HAL_DIR:PATH=OpenCV_HAL_DIR-NOTFOUND
OpenJPEG_DIR:PATH=/usr/lib64/openjpeg-2.4
PROTOBUF_UPDATE_FILES:BOOL=ON
PYTHON2_EXECUTABLE:FILEPATH=false
PYTHON2_INCLUDE_DIR:PATH=
PYTHON2_INCLUDE_DIR2:PATH=
PYTHON2_LIBRARY:FILEPATH=
PYTHON2_LIBRARY_DEBUG:FILEPATH=
PYTHON2_NUMPY_INCLUDE_DIRS:PATH=
PYTHON2_PACKAGES_PATH:PATH=
PYTHON3_EXECUTABLE:FILEPATH=/usr/bin/python3
PYTHON3_INCLUDE_DIR:PATH=/usr/include/python3.8
PYTHON3_INCLUDE_DIR2:PATH=
PYTHON3_LIBRARY:FILEPATH=/lib64/libpython3.8.so
PYTHON3_LIBRARY_DEBUG:FILEPATH=
PYTHON3_NUMPY_INCLUDE_DIRS:PATH=/usr/lib64/python3.8/site-packages/numpy/core/include
PYTHON3_PACKAGES_PATH:PATH=/usr/lib64/python3.8/site-packages
Pylint_DIR:PATH=Pylint_DIR-NOTFOUND
SYCL_DIR:PATH=SYCL_DIR-NOTFOUND
TBB_DIR:PATH=TBB_DIR-NOTFOUND
TBB_ENV_INCLUDE:PATH=/usr/include
TBB_ENV_LIB:FILEPATH=/lib64/libtbb.so
TBB_ENV_LIB_DEBUG:FILEPATH=TBB_ENV_LIB_DEBUG-NOTFOUND
TBB_VER_FILE:FILEPATH=/usr/include/tbb/tbb_stddef.h
Tesseract_DIR:PATH=Tesseract_DIR-NOTFOUND
VA_INCLUDE_DIR:PATH=/usr/include
VULKAN_INCLUDE_DIRS:PATH=/usr/include/vulkan
WITH_1394:BOOL=ON
WITH_ADE:BOOL=ON
WITH_ARAVIS:BOOL=OFF
WITH_CLP:BOOL=ON
WITH_CUDA:BOOL=OFF
WITH_EIGEN:BOOL=ON
WITH_FFMPEG:BOOL=OFF
WITH_FREETYPE:BOOL=OFF
WITH_GDAL:BOOL=ON
WITH_GDCM:BOOL=ON
WITH_GPHOTO2:BOOL=OFF
WITH_GSTREAMER:BOOL=ON
WITH_GTK:BOOL=ON
WITH_GTK_2_X:BOOL=OFF
WITH_HALIDE:BOOL=OFF
WITH_HPX:BOOL=OFF
WITH_IMGCODEC_HDR:BOOL=ON
WITH_IMGCODEC_PFM:BOOL=ON
WITH_IMGCODEC_PXM:BOOL=ON
WITH_IMGCODEC_SUNRASTER:BOOL=ON
WITH_INF_ENGINE:BOOL=OFF
WITH_IPP:BOOL=OFF
WITH_ITT:BOOL=OFF
WITH_JASPER:BOOL=ON
WITH_JPEG:BOOL=ON
WITH_JULIA:BOOL=OFF
WITH_LAPACK:BOOL=ON
WITH_LIBREALSENSE:BOOL=OFF
WITH_MATLAB:BOOL=OFF
WITH_MFX:BOOL=ON
WITH_NGRAPH:BOOL=OFF
WITH_ONNX:BOOL=OFF
WITH_OPENCL:BOOL=OFF
WITH_OPENCLAMDBLAS:BOOL=ON
WITH_OPENCLAMDFFT:BOOL=ON
WITH_OPENCL_SVM:BOOL=OFF
WITH_OPENEXR:BOOL=ON
WITH_OPENGL:BOOL=ON
WITH_OPENJPEG:BOOL=ON
WITH_OPENMP:BOOL=ON
WITH_OPENNI:BOOL=ON
WITH_OPENNI2:BOOL=OFF
WITH_OPENVX:BOOL=OFF
WITH_PLAIDML:BOOL=OFF
WITH_PNG:BOOL=ON
WITH_PROTOBUF:BOOL=ON
WITH_PTHREADS_PF:BOOL=ON
WITH_PVAPI:BOOL=OFF
WITH_QT:BOOL=OFF
WITH_QUIRC:BOOL=ON
WITH_TBB:BOOL=ON
WITH_TESSERACT:BOOL=ON
WITH_TIFF:BOOL=ON
WITH_UEYE:BOOL=OFF
WITH_V4L:BOOL=ON
WITH_VA:BOOL=ON
WITH_VA_INTEL:BOOL=OFF
WITH_VTK:BOOL=OFF
WITH_VULKAN:BOOL=ON
WITH_WEBP:BOOL=ON
WITH_XIMEA:BOOL=OFF
WITH_XINE:BOOL==OFF
dnnl_DIR:PATH=dnnl_DIR-NOTFOUND
opencv_dnn_PERF_CAFFE:BOOL=OFF
opencv_dnn_PERF_CLCAFFE:BOOL=OFF
```

--- a/modules/gapi/test/test_precomp.hpp
+++ b/modules/gapi/test/test_precomp.hpp
@@ -11,6 +11,7 @@
 #define __OPENCV_GAPI_TEST_PRECOMP_HPP__
 
 #include <cstdint>
+#include <thread>
 #include <vector>
 
 #include <opencv2/ts.hpp>

